mobify/platform-scaffold
# Fail gracefully throughout checkout when basket expires

This PR updates the commands on the checkout shipping and checkout payment pages to fail gracefully (ie. send the user to the homepage with a notification) when the cart has expired. 

 **JIRA**: https://mobify.atlassian.net/browse/WEBDATA-108
 **Linked PRs**: n/a

## Changes
- Wrapped checkout shipping and checkout payment init commands in container actions so we can handle expiry error messages
- Added expired basket error throwing to init checkout payment and checkout shipping SFCC commands
- Added expired basket error throwing to submit shipping and submit payment SFCC commands
- Added logic to handle submit shipping & submit payment commands throwing expired basket errors
- Added logic to handle expired basket errors in fetchShippingMethods action 

## How to test-drive this PR
- switch to the SFCC connector (in web/app/main.jsx)
- Run `npm run dev`
- Preview the [SFCC Site](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- add an item to the cart and proceed to the checkout shipping page
- wait on the page for a while so the cart will expire
- refresh the page
- you should be sent to the homepage with a notification stating your cart has expired
- add another item to the cart and proceed to the checkout shipping page
- fill out the form with valid data
- wait on the page for a while so the cart will expire
- try submitting the form to move to the payment page
- you should be sent to the homepage with a notification stating your cart has expired
- add another item to the cart and proceed to the checkout payment page
- wait on the page for the cart to expire
- refresh the page
- you should be sent to the homepage with a notification stating your cart has expired
- add another item to the cart and proceed to the checkout payment page
- fill out the form with valid information
- wait on the page for the cart to expire
- try submitting the order 
- you should be sent to the homepage with a notification stating your cart has expired
